fs/bcachefs/acl.c

```diff
@@ -369,8 +369,7 @@ int bch2_acl_chmod(struct btree_trans *trans,
 	xattr = bkey_s_c_to_xattr(bch2_btree_iter_peek_slot(iter));
 	acl = bch2_acl_from_disk(xattr_val(xattr.v),
 			le16_to_cpu(xattr.v->x_val_len));
-	ret = PTR_ERR_OR_ZERO(acl);
-	if (ret || !acl)
+	if (IS_ERR_OR_NULL(acl))
```

Comment:
This doesn't look right - you're losing the error code

Reply:
Ah! Good catch, not sure why I did that.